src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -982,6 +982,37 @@ class ArrayGenerator : public Generator<bsoncxx::array::value> {
     const UniqueGenerator<int64_t> _nTimesGen;
 };
 
+/** `{^Object: {key: {^RandomString: {length: 10}}, value: {^Inc: {}}, number: 10}` */
+class ObjectGenerator : public Generator<bsoncxx::document::value> {
+public:
+    ObjectGenerator(const Node& node,
+                   GeneratorArgs generatorArgs,
+                   std::map<std::string, Parser<UniqueAppendable>> parsers)
+        : _rng{generatorArgs.rng},
+          _node{node},
+          _generatorArgs{generatorArgs},
+          _keyGen{stringGenerator(node["key"], generatorArgs)},
+          _valueGen{valueGenerator<false, UniqueAppendable>(node["value"], generatorArgs, parsers)},
+          _nTimesGen{intGenerator(extract(node, "number", "^Object"), generatorArgs)} {}
```

Comment:
This is a great generator. Syntax 💅🏽 - please consider the following syntax / key names:

```python
{^Object: {withNEntries: 1000, havingKeys: {^Foo}, andValues: {^Bar}}}
```

Reply:
This syntax is not consistent with similar generator `^Array`, but I do not feel strongly about it. Done.